File: app/src/main/java/com/example/dell/chargetimer/settings/SharedPreferenceSettingsWriter.java
package com.example.dell.chargetimer.settings;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class SharedPreferenceSettingsWriter implements ISettingsWriter {
    private final SharedPreferences preferences;

    public SharedPreferenceSettingsWriter(Context context){
        preferences = PreferenceManager.getDefaultSharedPreferences(context);
    }

    @Override
    public void setFirstApplicationRunCompleted() {
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean("first_application_run", false);
        editor.apply();
    }

    @Override
    public void saveGoogleAdvancedNotificationsAllowed(boolean value){
        SharedPreferences.Editor editor = preferences.edit();
        editor.putBoolean("allow_calendar_permission_notifications", value);
        editor.apply();
    }
}

File: EvChargeTimer/app/src/main/java/com/chebuso/chargetimer/settings/SharedPreferenceSettingsWriter.kt
package com.chebuso.chargetimer.settings

import android.content.Context
import android.content.SharedPreferences
import androidx.preference.PreferenceManager

class SharedPreferenceSettingsWriter(context: Context): ISettingsWriter {
    private val context: Context
    private val preferences: SharedPreferences

    init {
        this.context = context
        this.preferences = PreferenceManager.getDefaultSharedPreferences(context)
    }

    override fun setFirstApplicationRunCompleted() {
        preferences.edit().apply{
            putBoolean("first_application_run", false)
        }.apply()
    }

    override fun saveCalendarAdvancedNotificationsAllowed(value: Boolean) {
        preferences.edit().apply{
            putBoolean("allow_calendar_permission_notifications", value)
        }.apply()
    }
}